integrationTests/vm/delegation/delegationScenarios_test.go

```diff
@@ -0,0 +1,480 @@
+package delegation
+
+import (
+	"bytes"
+	"encoding/binary"
+	"encoding/hex"
+	"math/big"
+	"strings"
+	"testing"
+
+	"github.com/ElrondNetwork/elrond-go/core"
+	"github.com/ElrondNetwork/elrond-go/crypto"
+	"github.com/ElrondNetwork/elrond-go/crypto/signing"
+	"github.com/ElrondNetwork/elrond-go/crypto/signing/mcl"
+	mclsig "github.com/ElrondNetwork/elrond-go/crypto/signing/mcl/singlesig"
+	"github.com/ElrondNetwork/elrond-go/data/block"
+	"github.com/ElrondNetwork/elrond-go/data/smartContractResult"
+	"github.com/ElrondNetwork/elrond-go/data/transaction"
+	"github.com/ElrondNetwork/elrond-go/integrationTests"
+	"github.com/ElrondNetwork/elrond-go/process/smartContract"
+	"github.com/ElrondNetwork/elrond-go/vm"
+	vmcommon "github.com/ElrondNetwork/elrond-vm-common"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestDelegationSystemNodesOperations(t *testing.T) {
```

Comment:
:hatoff: this would have taken at least 5 minutes to wait until completion. In this implementation it only took 40ms